packages/NodeTypeResolver/NodeTypeCorrector/PregMatchTypeCorrector.php

```diff
@@ -61,13 +59,11 @@ public function correct(Expr $expr, Type $originalType): Type
                 continue;
             }
 
-            if (! $this->argsAnalyzer->isArgInstanceInArgsPosition($funcCallNode->args, 2)) {
+            $thirdArg = $funcCallNode->getArgs()[2] ?? null;
```

Comment:
needs a count check on the number of args before, I guess?

Reply:
this cheap check should be moved before the `isNames()` call

Reply:
This does the same check actually